mantidproject/mantid
# EnggFocus: fix masking of bins - also fix Engg tests for GSL 1 and 2

The masking is now done at the beginning of the algorithm.

**To test:**
- All tests should pass
- As this requires updates in the tests, I made sure that the doc and system tests now pass with both GSL 1 and 2: http://builds.mantidproject.org/view/All/job/master_clean-archlinux/442/#showFailuresLink. I tested that on Ubuntu 16.04 (xenial).

Fixes #16883.
Also fixes #16678 (which is part of #16680).

There is [a small release note](https://github.com/mantidproject/mantid/commit/13d383fbf0830220c3276ab4109425340fb5cba9) for the change in EnggFocus.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
